File: nomad-samples/common/py/search_api/search_api/post_search.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def post_search(AUTH_TOKEN, PAGE_OFFSET, PAGE_SIZE, SEARCH_QUERY, FILTERS, SORT_FIELDS_NAME, 
                   SORT_FIELDS_ORDER, RESULT_FIELDS, IS_ADMIN) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = f"{ADMIN_API_URL}/admin/search" if IS_ADMIN else f"{PORTAL_API_URL}/portal/search"
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {"searchResultFields": RESULT_FIELDS}

    if PAGE_OFFSET != "": BODY["pageOffset"] = PAGE_OFFSET
    if PAGE_SIZE != "": BODY["pageSize"] = PAGE_SIZE
    if SEARCH_QUERY != "": BODY["searchQuery"] = SEARCH_QUERY 

    if len(FILTERS) != 0: BODY["filters"] = FILTERS

    if SORT_FIELDS_NAME != "": BODY["sortFields"] = [{"fieldName": SORT_FIELDS_NAME, "sortType": SORT_FIELDS_ORDER}]

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Searching failed")

File: nomad-samples/common/js/search-api/search-api/post-search.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function postSearch(AUTH_TOKEN, PAGE_OFFSET, PAGE_SIZE, SEARCH_QUERY, FILTERS, 
                                         RESULT_FIELDS_JSON, FIELD_NAME, SORT_TYPE, IS_ADMIN) 
{
    const API_URL = IS_ADMIN ? `${prjConstants.ADMIN_API_URL}/admin/search` : `${prjConstants.PORTAL_API_URL}/portal/search`;

    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    const BODY = {
        filters: FILTERS,
        SearchResultFields: RESULT_FIELDS_JSON
    };

    if (PAGE_OFFSET !== ""){BODY["pageOffset"] == PAGE_OFFSET}    
    if (PAGE_SIZE !== ""){BODY["pageSize"] == PAGE_SIZE}
    if (SEARCH_QUERY !== ""){BODY["searchQuery"] == SEARCH_QUERY}

    if (FIELD_NAME != "")
    {
        BODY.sortFields = [
            {
                fieldName: FIELD_NAME,
                sortType: SORT_TYPE
            }
        ]
    }

    // Post
    const RESPONSE = await fetch(API_URL, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Content search failed");
}